app/models/concerns/due_date_actions.rb

```diff
@@ -0,0 +1,300 @@
+# frozen_string_literal: true
+
+module DueDateActions
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :due_dates, as: :parent, dependent: :destroy
+    include DueDateQueries
+  end
+
+  # Generic activity permission checker that determines if an activity is permissible
+  # based on the current deadline state for this parent object
+  def activity_permissible?(activity)
+    current_deadline = next_due_date
+    return false unless current_deadline
+
+    current_deadline.activity_permissible?(activity)
+  end
+
+  # Syntactic sugar methods for common activities
+  # These provide clean, readable method names while avoiding DRY violations
+  def submission_permissible?
+    activity_permissible?(:submission)
+  end
+
+  def review_permissible?
+    activity_permissible?(:review)
+  end
+
+  def teammate_review_permissible?
+    activity_permissible?(:teammate_review)
+  end
+
+  def metareview_permissible?
+    activity_permissible?(:metareview)
+  end
+
+  def quiz_permissible?
+    activity_permissible?(:quiz)
+  end
+
+  def team_formation_permissible?
+    activity_permissible?(:team_formation)
+  end
+
+  def signup_permissible?
+    activity_permissible?(:signup)
+  end
+
+  def drop_topic_permissible?
+    activity_permissible?(:drop_topic)
+  end
+
+  # Check activity permissions for a specific deadline type
```

Comment:
This method should be removed completely.  The deadline type has nothing to do with whether an activity is permissible.  Whether an activity is permissible is entirely determined by the submission_allowed_id, review_allowed_id, etc.  Leaving this method in the code will simply confuse anyone who needs to maintain or extend this code.